src/stores/WalletStore.ts

```diff
@@ -38,24 +36,39 @@ class WalletStore extends PersistableStore {
     return web3Modal.cachedProvider
   }
 
+  private async getUserNetworkName() {
+    return (await provider.getNetwork()).name
+  }
+
+  private async isNetworkRight() {
+    const network = env.VITE_ETH_NETWORK
+    const userNetwork = await this.getUserNetworkName()
+    return network === userNetwork
+  }
+
   async connect(clearCachedProvider = false) {
     this.walletLoading = true
     try {
+      if (provider) provider.removeAllListeners()
       if (clearCachedProvider) web3Modal.clearCachedProvider()
 
       const instance = await web3Modal.connect()
-      provider = new Web3Provider(instance)
-
-      await this.checkNetwork(provider)
-      if (this.needNetworkChange)
-        throw new Error(
-          ErrorList.wrongNetwork(
-            (await provider.getNetwork()).name,
-            env.VITE_ETH_NETWORK
+      // We need this "any" networks so we could handle when users switch between networks with listeners without errors
+      provider = new Web3Provider(instance, 'any')
+
+      this.account = await this.getAccount()
+      if (!(await this.isNetworkRight())) {
```

Comment:
```TS
if (await this.isRightNetwork()) return
...
```
Less nesting = better readability (usually) + you don't need to convert boolean here, so  my code is a little faster

Reply:
But we don't want to return, we want to subscribe to our provider anyway, even if the network is not right

Reply:
I can see, then, extract the `await this.isRightNetwork()` into `const`, please, this will increase readability

Reply:
Got it. Sure, nice catch, ty